src/all/mangapluscreators/src/eu/kanade/tachiyomi/extension/all/mangapluscreators/MangaPlusCreators.kt

```diff
@@ -16,94 +18,200 @@ import okhttp3.Request
 import okhttp3.Response
 import rx.Observable
 import uy.kohesive.injekt.injectLazy
+import java.text.SimpleDateFormat
+import java.util.Locale
 
 class MangaPlusCreators(override val lang: String) : HttpSource() {
 
     override val name = "MANGA Plus Creators by SHUEISHA"
 
-    override val baseUrl = "https://medibang.com/mpc"
+    override val baseUrl = BASE_URL
 
     override val supportsLatest = true
 
     override fun headersBuilder(): Headers.Builder = Headers.Builder()
-        .add("Origin", baseUrl.substringBeforeLast("/"))
         .add("Referer", baseUrl)
         .add("User-Agent", USER_AGENT)
 
     private val json: Json by injectLazy()
 
+    // POPULAR Section
     override fun popularMangaRequest(page: Int): Request {
         val newHeaders = headersBuilder()
             .set("Referer", "$baseUrl/titles/popular/?p=m")
-            .add("X-Requested-With", "XMLHttpRequest")
             .build()
 
-        val apiUrl = "$API_URL/titles/popular/list".toHttpUrl().newBuilder()
-            .addQueryParameter("page", page.toString())
-            .addQueryParameter("pageSize", POPULAR_PAGE_SIZE)
-            .addQueryParameter("l", lang)
-            .addQueryParameter("p", "m")
-            .addQueryParameter("isWebview", "false")
-            .addQueryParameter("_", System.currentTimeMillis().toString())
-            .toString()
+        val popularUrl = "$baseUrl/titles/popular/?p=m&l=$lang".toHttpUrl().toString()
 
-        return GET(apiUrl, newHeaders)
+        return GET(popularUrl, newHeaders)
     }
 
-    override fun popularMangaParse(response: Response): MangasPage {
-        val result = response.asMpcResponse()
+    override fun popularMangaParse(response: Response): MangasPage = parseMangasPageFromElement(
+        response,
+        "div.item-recent",
+    )
 
-        checkNotNull(result.titles) { EMPTY_RESPONSE_ERROR }
+    fun parseMangasPageFromElement(response: Response, selector: String): MangasPage {
```

Comment:
make these functions private

Reply:
what about the classes?
https://github.com/curche/keiyoushi-extensions-source/blob/mpc-jp/src/all/mangapluscreators/src/eu/kanade/tachiyomi/extension/all/mangapluscreators/MangaPlusCreators.kt#L352

Reply:
> what about the classes?

Make them `private` if you feels like it. Not a big problem.